kart/tabular/v3.py

```diff
@@ -60,43 +82,68 @@ class TableV3(RichTableDataset):
 
     DATASET_DIRNAME = ".table-dataset"  # New name for V3 datasets.
 
-    # All relative paths should be relative to self.inner_tree - that is, to the tree named DATASET_DIRNAME.
-    FEATURE_PATH = "feature/"
     META_PATH = "meta/"
-
     LEGEND_DIRNAME = "legend"
     LEGEND_PATH = META_PATH + "legend/"
     SCHEMA_PATH = META_PATH + "schema.json"
 
-    TITLE_PATH = META_PATH + "title"
-    DESCRIPTION_PATH = META_PATH + "description"
-
-    CRS_PATH = META_PATH + "crs/"
-
-    PATH_STRUCTURE_PATH = META_PATH + "path-structure.json"
-    CAPABILITIES_PATH = META_PATH + "capabilities.json"
-
-    # Attachments
-    METADATA_XML = "metadata.xml"
+    # === Visible meta-items ===
+    # Override schema.json to normalise schemas during serialisation.
+    SCHEMA_JSON = MetaItemDefinition("schema.json", SchemaJsonFileType.INSTANCE)
+
+    # == Hidden meta-items (which don't show in diffs) ==
+    # How automatically generated PKs have been assigned so far:
+    GENERATED_PKS = MetaItemDefinition(
+        "generated-pks.json", MetaItemFileType.JSON, MetaItemPermission.HIDDEN
+    )
+    # How primary keys are converted to feature paths:
+    PATH_STRUCTURE = MetaItemDefinition(
+        "path-structure.json", MetaItemFileType.JSON, MetaItemPermission.INTERNAL_ONLY
+    )
+    # Legends are used to help decode each feature:
+    LEGEND = MetaItemDefinition(
+        re.compile(r"legend/(.*)"),
+        MetaItemFileType.BYTES,
+        MetaItemPermission.INTERNAL_ONLY,
+    )
+
+    META_ITEMS = (
+        BaseDataset.TITLE,
+        BaseDataset.DESCRIPTION,
+        BaseDataset.METADATA_XML,
+        SCHEMA_JSON,
+        BaseDataset.CRS_DEFINITIONS,
+        GENERATED_PKS,
+        PATH_STRUCTURE,
+        LEGEND,
+    )
+
+    # This meta-item is generally stored in the "attachment" area, alongside the dataset, rather than inside it.
+    # Storing it in this unusual location adds complexity without actually solving any problems, so newer datasets
+    # don't do this.
```

Comment:
in table dataset v3 we still do this, so maybe clarify what you mean by 'newer datasets' (point clouds)

Reply:
I've added "for example point clouds", so I don't have to maintain a list here of datasets newer than table.v3